code/src/main/java/com/onlinecontacttracing/messaging/SendEmails.java

```diff
@@ -0,0 +1,94 @@
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+
+public class SendEmails {
+
+  private String emailSubject;
+  private String emailBody;
+  private MessageServlet messageObject;
+  private ArrayList<PotentialContact> contactsList;
+  private Gmail service;
+  private PositiveUser user;
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String TOKENS_DIRECTORY_PATH = "tokens";
+  private static final List<String> SCOPES = Collections.singletonList(GmailScopes.GMAIL_SEND);
+  private static final String CREDENTIALS_FILE_PATH = "/credentials.json";//rename to be likek authconfigurationdata, also need to remove from github
+  // final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+  
+  public SendEmails (SystemMessage systemMessage, LocalityResource localityResource, CustomizableMessage customizableMessage, String emailSubject, String messageLanguage, ArrayList<PotentialContact> contactsList, PositiveUser user) {
+    messageObject = new MessageServlet(systemMessage, localityResource, customizableMessage, user);
+    emailBody = messageObject.compileMessage(messageObject.statusListToShowUser(messageLanguage));
+    this.emailSubject= emailSubject;
+    this.contactsList = contactsList;
+    // service = new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, getCredentials(HTTP_TRANSPORT, user))
+    //             .setApplicationName(APPLICATION_NAME)
+    //             .build();
+    this.user = user;
+  }
+
+  public void sendEmailsOut () {
+    PotentialContact contact;
+    MimeMessage email;
+    for(PotentialContact contactName : contactsList) {
+      // email = MessagingSetup.createEmail(contactName.getEmail(), user.getUserEmail(), emailSubject, emailBody);
+      // sendMessage(service, user.getUserId(), email);
+    }
+  }
+
+  private static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT, PositiveUser user) throws IOException {
+        // Load client secrets
+        InputStream in = SendEmails.class.getResourceAsStream(CREDENTIALS_FILE_PATH);//create a class authAPI config constants, have this as a string, eliminate input reader
+        if (in == null) {
+            throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+        }
+        GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+      
+        TokenResponse tokenResponse = new TokenResponse().setScope(SCOPES.get(0));
+        return new GoogleAuthorizationCodeFlow.Builder(
+               HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES).build().createAndStoreCredential(tokenResponse, user.getUserId());
+        //  = new GoogleAuthorizationCodeFlow.Builder(
```

Comment:
Presumably you're still working on this?

Reply:
yup, this is part of the "need service account" section